test/integration/good/code-gen/opencl/cpp.expected

```diff
@@ -6729,38 +6744,43 @@ class restricted_model final : public model_base_crtp<restricted_model> {
       p_real = std::numeric_limits<double>::quiet_NaN();
       
       current_statement__ = 1;
-      p_real = in__.scalar();
+      p_real = in__.read<local_scalar_t__>(lp__);
```

Comment:
Close! Just gotta get rid of the `lp__` for the `read()` methods

Reply:
woooooops